### PR TITLE
Explorer search match hex values whether it has a 0x prefix or not

### PR DESCRIPTION
**Pull request by NavyAdmiral**
For [#165976020](https://www.pivotaltracker.com/story/show/165976020)

### CODE REVIEW

File: explorer/src/queries/search.ts [hunk 1]
```diff
@@ -15,12 +21,21 @@ const searchBuilder = (
 
   if (params.searchQuery != null) {
     const searchTokens = params.searchQuery.split(/\s+/)
+    const normalizedSearchTokens = searchTokens.map(token =>
```
Comment (rupurt) [inline explorer/src/queries/search.ts]: I think this can be reduced to `searchTokens(normalizeSearchToken)`

File: explorer/src/queries/search.ts [hunk 2]
```diff
@@ -7,6 +7,12 @@ export interface ISearchParams {
   limit?: number
 }
 
+const normalizeSearchToken = (id: string): string => {
+  if (id && id.substr(0, 2) !== '0x' && (id.length === 64 || id.length === 40))
```
Comment (rupurt) [inline explorer/src/queries/search.ts]: How do you feel about extracting the magic numbers `64` & `40` into meaningful const variables?

File: explorer/src/queries/search.ts (hunk 1 above)
Comment (NavyAdmiral) [reply explorer/src/queries/search.ts]: Isn't searchTokens an array and normalizeSearchToken a function that acts on a single string?

Comment (rupurt) [reply explorer/src/queries/search.ts]: Sorry. That's my bad. I meant

```
searchTokens.map(normalizeSearchToken)
```

File: explorer/src/queries/search.ts (hunk 2 above)
Comment (NavyAdmiral) [reply explorer/src/queries/search.ts]: I agree. Do you like the introduced names?

Comment (rupurt) [reply explorer/src/queries/search.ts]: Looking good. Just a couple of linting errors

Comment (NavyAdmiral) [reply explorer/src/queries/search.ts]: Need to make a precommit hook or else they're getting out of hand.
@rupurt Don't we want to support fuzzy searching? In my opinion a basic `like %%` to display results for substring queries wouldn't be bad

Comment (rupurt) [reply explorer/src/queries/search.ts]: We only want to support exact match searching at this point.

**Pull request merged**